src/main/java/com/fasterxml/uuid/impl/LazyRandom.java

```diff
@@ -9,9 +9,26 @@
  */
 public final class LazyRandom
 {
-    private final static SecureRandom shared = new SecureRandom();
+    private static final Object lock = new Object();
+    private static volatile SecureRandom shared;
 
     public static SecureRandom sharedSecureRandom() {
-        return shared;
+        // Double check lazy initialization idiom (Effective Java 3rd edition item 11.6)
```

Comment:
Since this is not called very often, I think I'll modify this to use straight synchronization instead of double-locking.